### ISSUE TITLE
Querying for document by _key returning only a single seemingly random property on entity ("old", in this case)

### ISSUE THREAD

**Issue by lostpebble** [arangodb/arangodb#11590]
## My Environment

* __ArangoDB Version__:        3.6.3
* __Storage Engine__:             RocksDB

## Component, Query & Data

__Affected feature__:

ArangoDB Web Interface

__Problem__:

I've created a new collection, and then added a bunch of documents to it. But when I try and view individual documents in the web view by clicking on them in the collection, I'm not getting anything displaying:

![image](https://user-images.githubusercontent.com/1508863/81670789-68800d00-9448-11ea-9245-f748f4123327.png)

I opened up the browser's console and looked at the network requests. This is the document request, which uses `/lookup-by-keys`:

```json
{"keys":["za_emalahleni"],"collection":"old_magnetic_to_geolevel"}
```

And this is the response:

```json
{"documents":["za/emalahleni"],"error":false,"code":200}
```

Whereas on a different collection, which is working, the request looks like this:

```json
{"keys":["us_berea"],"collection":"old_magnetic_blueprints"}
```

And the response is correct:

```json
{"documents":[{"_key":"us_berea","_id":"old_magnetic_blueprints/us_berea","_rev":"_adOCmfa---","urlIncludesRegion":false,"coordinates":{"latitude":41.36616134643555,"longitude":-81.85430145263672},"regionName":"Ohio","localityName":"Berea","overriddenSearchRadius":0,"locationLevel":"LOCALITY","regionCode":"OH","countryCode":"US","overriddenCoordinates":null,"subLocalityName":null,"dateLastUpdated":"2017-02-08T16:55:10.101Z","pageUrl":"us/berea","placeId":"ChIJie4S0b2UMIgRvqvkpHc6qh4","id":"us/berea"}],"error":false,"code":200}
```

__Expected result__:

The web view shows the entire document as expected.

I can still run a query against this dataset and get a full correctly formatted table back, so as far as I can tell there is no difference between these two collections.

```
FOR doc IN old_magnetic_blueprints
  LIMIT 1
  RETURN doc
```

```json
[
  {
    "_key": "us_berea",
    "_id": "old_magnetic_blueprints/us_berea",
    "_rev": "_adOCmfa---",
    "urlIncludesRegion": false,
    "coordinates": {
      "latitude": 41.36616134643555,
      "longitude": -81.85430145263672
    },
    "regionName": "Ohio",
    "localityName": "Berea",
    "overriddenSearchRadius": 0,
    "locationLevel": "LOCALITY",
    "regionCode": "OH",
    "countryCode": "US",
    "overriddenCoordinates": null,
    "subLocalityName": null,
    "dateLastUpdated": "2017-02-08T16:55:10.101Z",
    "pageUrl": "us/berea",
    "placeId": "ChIJie4S0b2UMIgRvqvkpHc6qh4",
    "id": "us/berea"
  }
]
```

And the exact same query, but using the collection which isn't displaying properly in the UI:

```
FOR doc IN old_magnetic_to_geolevel
  LIMIT 1
  RETURN doc
```

```json
[
  {
    "_key": "za_emalahleni",
    "_id": "old_magnetic_to_geolevel/za_emalahleni",
    "_rev": "_adS8qBq--p",
    "old": "za/emalahleni",
    "oldLocationLevel": "LOCALITY",
    "new": "za(mp)emalahleni",
    "shortLvl": "inapplicable",
    "tagGeo": [],
    "distance": 18645,
    "lvlValue": 60,
    "lvlType": "local_municipality",
    "name": "Emalahleni",
    "geoLvlShort": "za(mp)emalahleni"
  }
]
```

The documents internally seem to look exactly the same, so I'm unsure why `/lookup-by-keys`, which the web interface seems to use to pull the individual docs, would not work the same.

---

Oh, okay.. So I just tried a query that filter's by key - to replicate `/lookup-by-keys` exactly, and the results are interesting:

```
FOR doc IN old_magnetic_to_geolevel
  FILTER doc._key == "za_emalahleni"
  RETURN doc
```

```json
[
  "za/emalahleni"
]
```

So this is consistent with what we're getting in the interface now. This surely can't be regular behavior though? Even what's being returned is weird, as its just returning the value set at the property `"old": "za/emalahleni"`.

**Comment by jsteemann** [comment]
Looks strange indeed. Just wondering if this is a cluster or a single server?
Can you also provide the query execution plan (i.e. explain output) for the query that does not return the expected result? Thanks!

**Comment by lostpebble** [comment]
This is running on ArangoDB Oasis, here's a quick copy-paste of the config:

VERSION
3.6.3
MODEL
OneShard
NODES MEMORY
3 × 4 GB
NODES CPU
General
NODES DISK
3 × 10 GB

And here's the query execution:

```
Query String (85 chars, cacheable: true):
 FOR doc IN old_magnetic_to_geolevel
   FILTER doc._key == "za_emalahleni"
   RETURN doc

Execution plan:
 Id   NodeType                    Site  Est.   Comment
  1   SingletonNode               COOR     1   * ROOT
  7   SingleRemoteOperationNode   COOR     1     - FOR doc IN old_magnetic_to_geolevel FILTER _key == "za_emalahleni" /* primary index scan */
  5   ReturnNode                  COOR     1     - RETURN doc

Indexes used:
 By   Name      Type      Collection                 Unique   Sparse   Selectivity   Fields       Ranges
  7   primary   primary   old_magnetic_to_geolevel   true     false       100.00 %   [ `_key` ]   "za_emalahleni"

Optimization rules applied:
 Id   RuleName
  1   use-indexes
  2   remove-filter-covered-by-index
  3   remove-unnecessary-calculations-2
  4   optimize-cluster-single-document-operations
```

**Comment by jsteemann** [comment]
I was able to reproduce the issue and fixed it via the above-mentioned PRs.

**Comment by maxkernbach** [comment]
Hi @bischoje,

Version 3.6.4, which includes a fix for this issue, has been released.

From the [Changelog](https://github.com/arangodb/arangodb/blob/25e711db88b5f43915cf61b2e895fe9e86b35051/CHANGELOG#L99):

> 
> * Fixed issue #11590: Querying for document by _key returning only a single
>   seemingly random property on entity ("old", in this case).
> 
>   This fixes single-key document lookups in the cluster for simple by-key
>   AQL queries, such as `FOR doc IN collection FILTER doc._key == @key RETURN
>   doc` in case the document has either an "old" or a "new" attribute.

Please find the latest release in the [download section](https://www.arangodb.com/download/).
